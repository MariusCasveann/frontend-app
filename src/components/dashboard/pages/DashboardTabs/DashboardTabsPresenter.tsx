import * as React from 'react';

// components
import GovernanceDashboardPresenter from '../GovernanceDashboard/GovernanceDashboardPresenter';
import { TabProps } from '../../../common/molecules/Tabs/TabsPresenter';
import { TabsPresenter } from '../../../common/molecules/Tabs';

// style
import './DashboardTabs.css';
import DomainDashboardPresenter from '../DomainDashboard/DomainDashboardPresenter';
import ProductDashboardPresenter from '../ProductDashboard/ProductDashboardPresenter';

const tabs: TabProps[] = [
    {
        tabContent: GovernanceDashboardPresenter,
        tabName: 'Governance'
    },
    {
        tabContent: DomainDashboardPresenter,
        tabName: 'Domain'
    },
    {
        tabContent: ProductDashboardPresenter,
        tabName: 'Product'
    }
];

export default () => (
    <div className="dashboard-page">
        <TabsPresenter tabsData={tabs} />
    </div>
);
