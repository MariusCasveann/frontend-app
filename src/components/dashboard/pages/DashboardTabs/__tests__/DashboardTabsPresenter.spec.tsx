import * as React from 'react';

jest.mock(
    '../../../../dashboard/pages/GovernanceDashboard/GovernanceDashboardPresenter',
    () => 'GovernanceDashboardPresenter'
);
jest.mock('../../../../common/molecules/Tabs/TabsPresenter', () => ({ TabProps: 'TabProps' }));
jest.mock('../../../../common/molecules/Tabs', () => ({ TabsPresenter: 'TabsPresenter' }));

import DashboardTabsPresenter from './../DashboardTabsPresenter';
import * as ShallowRenderer from 'react-test-renderer/shallow';

describe('DashboardTabsPresenter', () => {
    it('renders correctly', () => {
        const renderer = ShallowRenderer.createRenderer();
        renderer.render(<DashboardTabsPresenter />);
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});
