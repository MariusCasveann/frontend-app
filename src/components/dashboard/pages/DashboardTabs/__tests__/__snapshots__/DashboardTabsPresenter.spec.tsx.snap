// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTabsPresenter renders correctly 1`] = `
<div
  className="dashboard-page"
>
  <TabsPresenter
    tabsData={
      Array [
        Object {
          "tabContent": "GovernanceDashboardPresenter",
          "tabName": "Governance",
        },
        Object {
          "tabContent": [Function],
          "tabName": "Domain",
        },
        Object {
          "tabContent": [Function],
          "tabName": "Product",
        },
      ]
    }
  />
</div>
`;
